<!-- nav_goal_recorder.launch -->
<launch>
  <!-- Path to save the recorded poses CSV file -->
  <arg name="save_dir" default="$(find open_autonomous_tractor)/datasets/nav_goals" />
  <arg name="csv_filename" default="recorded_poses.csv" />
  <arg name="frame_id" default="map" />
  
  <!-- Run the nav goal recorder node -->
  <node pkg="open_autonomous_tractor" type="nav_goal_recorder.py" name="nav_goal_recorder" output="screen">
    <param name="save_dir" value="$(arg save_dir)" />
    <param name="csv_filename" value="$(arg csv_filename)" />
    <param name="frame_id" value="$(arg frame_id)" />
  </node>
</launch>